src/oop/programs_with_multiple_classes/medium/q3/MainTest.java

```diff
@@ -0,0 +1,85 @@
+package oop.programs_with_multiple_classes.medium.q3;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String classPencil = "Pencil";
+    private final String classEraser = "Eraser";
+    private final String varSize = "size";
+    private final String varHardness = "hardness";
+    private final String varSchool = "isForSchool";
+    public ObjectTest pencil;
+    public ObjectTest eraser;
+
+    @BeforeEach
+    public void setup() {
+        String pencilClassString = "oop.programs_with_multiple_classes.medium.q3." + classPencil;
+        String eraserClassString = "oop.programs_with_multiple_classes.medium.q3." + classEraser;
+        pencil = new ObjectTest(pencilClassString);
+        eraser = new ObjectTest(eraserClassString);
+        String modifiedPencilMessage =
+                "You have modified the class fields in the " + classPencil + " class. Please revert them back to the original state they were provided in.";
+        assertTrue(pencil.hasField(varHardness, int.class), modifiedPencilMessage);
+        assertTrue(pencil.hasField(varSchool, boolean.class), modifiedPencilMessage);
+        assertTrue(pencil.hasModifier(varHardness, "private"), modifiedPencilMessage);
+        assertTrue(pencil.hasModifier(varSchool, "private"), modifiedPencilMessage);
+        String modifiedEraserMessage =
+                "You have modified the class fields in the " + classEraser + " class. Please revert them back to the original state they were provided in.";
```

Comment:
They were provided with the constructor here, not the fields. I think we should develop a standard error message for this sort of thing

Reply:
Maybe `You have modified the provided portions of class x. Please revert them to the original state.`